certification/internal/authn/keychain.go

```diff
@@ -0,0 +1,68 @@
+package authn
+
+import (
+	"os"
+	"sync"
+
+	"github.com/docker/cli/cli/config"
+	"github.com/docker/cli/cli/config/types"
+	craneauthn "github.com/google/go-containerregistry/pkg/authn"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+type preflightKeychain struct {
+	mu sync.Mutex
+}
+
+var PreflightKeychain craneauthn.Keychain = &preflightKeychain{}
+
+func (k *preflightKeychain) Resolve(target craneauthn.Resource) (craneauthn.Authenticator, error) {
```

Comment:
I see that this resembles `main` of `crane` which I think is smart, however in the version we are referencing as a dependency the code looks a little different, see [here](https://github.com/google/go-containerregistry/blob/v0.8.0/pkg/authn/keychain.go#L119-L122).

Specifically, I've experience this weird issue (for a lack of a better term) with preflight and docker registry. Do you think we need to account for the above in our code as well? Or do you feel since `main` got ride of this code that it's handled elsewhere in their dependencies and we should also be fine? Or should we _also_ update preflight to use `main` so it's more of a one-to-one? If this is unclear please let me know.

Reply:
Added.